### PR TITLE
Add Rawls endpoints for GAWB-3293 testing

### PR DESCRIPTION
**Pull request by jmthibault79**
**PR checklist**
- [ ] For each library you've modified here, decide whether it requires a major, minor, or no version bump. (Click [here](/broadinstitute/workbench-libs/blob/develop/CONTRIBUTING.md) for guidance)

If you're doing a **major** or **minor** version bump to any libraries:
- [ ] Bump the version in `project/Settings.scala` `createVersion()`
- [ ] Update `CHANGELOG.md` for those libraries
- [ ] I promise I used `@deprecated` instead of deleting code where possible

In all cases:
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green (CI _and_ coverage tests)
- [ ] Squash commits and **merge to develop**
- [ ] Delete branch after merge

After merging, _even if you haven't bumped the version_: 
- [ ] Update `README.md` and the `CHANGELOG.md` for any libs you changed with the new dependency string. The git hash is the same short version you see in GitHub (i.e. seven characters). You can commit these changes straight to develop.

### CODE REVIEW

**Review by aweng98 — COMMENTED** [review]
trait `BillingFixtures.makeRandomId` is `RandomUtil`, right?

**Comment by jmthibault79** [comment]
Some klugey API parsing here, to enable only what's necessary for my ticket.

The "right" way for tests to interact with APIs is worth a discussion and will be captured in GAWB-3319.

File: serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/service/Rawls.scala [hunk 1]
```diff
@@ -50,6 +53,58 @@ trait Rawls extends RestClient with LazyLogging {
       parseResponse(getRequest(url + s"api/workspaces"))
     }
   }
+
+  object submissions {
+    // dear reader: "workspace namespace" is an archaic term for billing project which is still used in a few places
```
Comment (breilly2) [inline serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/service/Rawls.scala]: Then we should start using the preferred term in new code.

File: serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/fixture/MethodData.scala [hunk 2]
```diff
@@ -45,9 +47,168 @@ object MethodData {
     documentation = "",
     payload = "task hello {\n  String name\n\n  command {\n    echo 'hello ${name}!'\n  }\n  output {\n    File response = stdout()\n  }\n  runtime {\n    docker: \"ubuntu\"\n  }\n}\n\nworkflow test {\n  call hello\n}"
   )
+
+  val SubWorkflowChildMethod = Method(
```
Comment (breilly2) [inline serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/fixture/MethodData.scala]: This stuff is probably needed for a specific test, right? If so, it should be closer to that test. It bothered me a little when `MethodData` was in the firecloud-ui repo but it bothers me a lot now that it is in workbench-libs.

**Review by breilly2 — CHANGES_REQUESTED** [review]
Unless there's a really good reason to have it here, I want to see test data closer to the test.

File: serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/fixture/MethodData.scala (hunk 2 above)
Comment (helgridly) [reply serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/fixture/MethodData.scala]: I am inclined to agree. If it's used in more than one test, it should be here; if not, it should be with the test.

Offtopic/related/the other side of the same coin: it'd be really nice to have some fixtures that the test environment spins up with (like our user pool). Workspaces and such.

Comment (jmthibault79) [reply serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/fixture/MethodData.scala]: Makes sense.  I'll move it to the test.

File: serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/service/Rawls.scala (hunk 1 above)
Comment (jmthibault79) [reply serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/service/Rawls.scala]: true

**Review by breilly2 — APPROVED** [review]
👍 Thanks for making the change!

File: serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/service/Rawls.scala [hunk 3]
```diff
@@ -50,6 +54,56 @@ trait Rawls extends RestClient with LazyLogging {
       parseResponse(getRequest(url + s"api/workspaces"))
     }
   }
+
+  object submissions {
+    def launchWorkflow(billingProject: String, workspaceName: String, methodConfigurationNamespace: String, methodConfigurationName: String, entityType: String, entityName: String, expression: String, useCallCache: Boolean, workflowFailureMode: String = "NoNewCalls")(implicit token: AuthToken): String = {
+      val body = Map("methodConfigurationNamespace" -> methodConfigurationNamespace, "methodConfigurationName" -> methodConfigurationName, "entityType" -> entityType, "entityName" -> entityName, "expression" -> expression, "useCallCache" -> useCallCache, "workflowFailureMode" -> workflowFailureMode)
+      logger.info(s"Creating a submission: $billingProject/$workspaceName config: $methodConfigurationNamespace/$methodConfigurationName with body $body")
+      val response = postRequest(url + s"api/workspaces/$billingProject/$workspaceName/submissions", body)
```
Comment (aweng98) [inline serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/service/Rawls.scala:62]: All good, only a short comment on the placement of log. I wish everyone logs action after action took place as the standard practice.

**Review by aweng98 — APPROVED** [review]

**Review by helgridly — APPROVED** [review]

File: serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/service/Rawls.scala (hunk 3 above)
Comment (jmthibault79) [reply serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/service/Rawls.scala:62]: Disagree.  Logging-after means that there's no log when the action fails.

**Pull request merged**